### PR TITLE
cleanup / add bazel canary jobs

### PR DESCRIPTION
**Pull request by BenTheElder**
This also tags the latest and greatest bazel-version-image with `latest-latest` so the canary jobs can always use that tag.

### CODE REVIEW

**Comment by BenTheElder** [comment]
/area bazel
/area jobs
/assign @krzyzacy 
cc @ixdy

File: prow/config.yaml [hunk 1]
```diff
@@ -606,6 +606,55 @@ presubmits:
...
           hostPath:
             path: /mnt/disks/ssd0
+
+  - name: pull-kubernetes-bazel-build-canary
+    agent: kubernetes
+    context: pull-kubernetes-bazel-build-canary
+    always_run: false
+    rerun_command: "/test pull-kubernetes-bazel-build-canary"
+    trigger: "(?m)^/test pull-kubernetes-bazel-build-canary,?(\\s+|$)"
+    skip_branches:
```
Comment (krzyzacy) [inline prow/config.yaml]: nit: also `branches: master` to unify with the test job?

Comment (BenTheElder) [reply prow/config.yaml]: thanks, fixed!

File: images/bazelbuild/Makefile [hunk 2]
```diff
@@ -27,7 +29,10 @@ do_build = \
 do_push = \
 	gcloud docker -- push $(IMG):$(TAG)-$(VERSION);\
 	gcloud docker -- push $(IMG):latest-$(VERSION);\
-	echo Pushed $(IMG) with $(TAG)-$(VERSION) and latest-$(VERSION) tags;
+	echo Pushed $(IMG) with $(TAG)-$(VERSION) and latest-$(VERSION) tags;\
+	[[ $(VERSION) == $(LATEST_BAZEL_VERSION) ]];\
```
Comment (krzyzacy) [inline images/bazelbuild/Makefile]: maybe you need `ifeq`? I don't see a `latest-latest` tag :-)

Comment (BenTheElder) [reply images/bazelbuild/Makefile]: this works now, I wasn't tagging the image :confused:

**Comment by krzyzacy** [comment]
lgtm after fix presubmits :-)

**Comment by BenTheElder** [comment]
I forgot testgrid :-(

**Pull request merged**